compiler/rustc_typeck/src/check/op.rs

```diff
@@ -505,6 +505,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             }
             // We're emitting a suggestion, so we can just ignore regions
             let fn_sig = self.tcx.fn_sig(def_id).skip_binder();
+            if fn_sig.has_escaping_bound_vars() {
```

Comment:
^ and removing the two lines above.
```suggestion
            // FIXME: Instead of exiting early when encountering bound vars in the function
            // signature, consider keeping the binder here and propagating this downwards.
            let fn_sig = if let Some(fn_sig) = self.tcx.fn_sig(def_id).no_bound_vars() {
                 fn_sig
            } else {
                return false;
            }
```

Reply:
@ldm0 this fixme is either something you can do in this PR or something to potentially do in a followup, whatever you prefer.

Reply:
Thanks! I thinks it's better to fix this ICE in a small way.